x-pack/test/cases_api_integration/common/lib/authentication/index.ts

```diff
@@ -20,9 +20,8 @@ export const getUserInfo = (user: User): UserInfo => ({
 
 export const createSpaces = async (getService: CommonFtrProviderContext['getService']) => {
   const spacesService = getService('spaces');
-  for (const space of spaces) {
-    await spacesService.create(space);
-  }
+
+  await Promise.all(spaces.map((space) => spacesService.create(space)));
```

Comment:
Made some optimizations to speed up the creation of spaces, roles, and users for the tests